receiver/jmxreceiver/internal/subprocess/subprocess.go

```diff
@@ -285,11 +291,42 @@ func collectStdout(stdoutScanner *bufio.Scanner, stdoutChan chan<- string, logge
 		if text != "" {
 			stdoutChan <- text
 			logger.Debug(text)
+
+			if statusCode := checkForStatusCode(text); statusCode != "" {
+				logger.Error(text, zap.String("status_code", statusCode))
+			}
 		}
 	}
 	// Returns when stdout is closed when the process ends
 }
 
+func checkForStatusCode(text string) string {
+	if strings.Contains(text, "Could not connect to remote JMX server") {
+		return "UNAVAILABLE"
+	} else if strings.Contains(text, "`interval` must be positive:") {
+		return codes.InvalidArgument.String()
+	} else if strings.Contains(text, "`otlp.timeout` must be positive:") {
+		return codes.InvalidArgument.String()
+	} else if strings.Contains(text, "jmx missing required fields:") {
+		return codes.InvalidArgument.String()
+	} else if strings.Contains(text, "failed to parse OTLPExporterConfig.Endpoint") {
+		return codes.FailedPrecondition.String()
+	} else if strings.Contains(text, "failed to parse Endpoint") {
+		return codes.FailedPrecondition.String()
+	} else if strings.Contains(text, "no subprocess.cancel().") {
+		return codes.FailedPrecondition.String()
```

Comment:
How does this happen? Seems like an internal error? Same w/ the last two subprocess errors.

Reply:
yeah, I honestly thought so too but was hesitant to use internal error because I thought it was meant for more serious errors.